File: lib_common/src/main/java/com/btg/common/base/BaseApplication.java
package com.btg.common.base;

import android.app.Application;

import androidx.annotation.Nullable;

import com.btg.common.BuildConfig;
import com.orhanobut.logger.AndroidLogAdapter;
import com.orhanobut.logger.FormatStrategy;
import com.orhanobut.logger.LogcatLogStrategy;
import com.orhanobut.logger.Logger;
import com.orhanobut.logger.PrettyFormatStrategy;

/**
 * @创建者 567
 * @创建时间 2020/8/18 5:07 PM
 * @描述 Application 基类
 */
public class BaseApplication extends Application {

    private static Application sApplication;

    public static Application getApplication() {
        return sApplication;
    }

    @Override
    public void onCreate() {
        super.onCreate();
        sApplication = this;

        FormatStrategy formatStrategy = PrettyFormatStrategy.newBuilder()
                .showThreadInfo(false)  // 是否显示线程信息 默认显示 上图Thread Infrom的位置
                .methodCount(0)         // 展示方法的行数 默认是2  上图Method的行数
                .methodOffset(7)        // 内部方法调用向上偏移的行数 默认是0
                .logStrategy(new LogcatLogStrategy()) // 改变log打印的策略一种是写本地，一种是logcat显示 默认是后者（当然也可以自己定义）
                .tag("BTG_LOG")   // 自定义全局tag 默认：PRETTY_LOGGER
                .build();
        Logger.addLogAdapter(new AndroidLogAdapter() {
            @Override
            public boolean isLoggable(int priority, @Nullable String tag) {
                return BuildConfig.DEBUG;
            }
        });

    }
}

File: lib_common/src/main/java/com/btg/common/base/BaseApplication.kt
package com.btg.common.base

import android.app.Application
import com.btg.common.BuildConfig
import com.btg.common.app.AppForegroundObserver
import com.btg.common.app.CrashHandler
import com.orhanobut.logger.AndroidLogAdapter
import com.orhanobut.logger.Logger
import com.orhanobut.logger.PrettyFormatStrategy

/**
 * Application 基类：统一初始化日志、全局崩溃捕获、App 前后台监听。
 * 具体 app 的 Application 继承它（并加 @HiltAndroidApp），在 manifest 用 android:name 注册。
 */
open class BaseApplication : Application() {

    override fun onCreate() {
        super.onCreate()
        instance = this
        initLogger()
        CrashHandler.install()
        AppForegroundObserver.init()
    }

    private fun initLogger() {
        val formatStrategy = PrettyFormatStrategy.newBuilder()
            .showThreadInfo(false)
            .methodCount(0)
            .methodOffset(7)
            .tag("BTG_LOG")
            .build()
        Logger.addLogAdapter(object : AndroidLogAdapter(formatStrategy) {
            override fun isLoggable(priority: Int, tag: String?): Boolean = BuildConfig.DEBUG
        })
    }

    companion object {
        /** 全局 Application 实例（onCreate 后可用）。 */
        lateinit var instance: BaseApplication
            private set
    }
}
